hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/testutils/SparkDatasetTestUtils.java

```diff
@@ -173,4 +176,17 @@ public static InternalRow getInternalRowWithError(String partitionPath) {
         .withBulkInsertParallelism(2);
   }
 
+  private static InternalRow serializeRow(ExpressionEncoder encoder, Row row)
+      throws InvocationTargetException, IllegalAccessException, NoSuchMethodException, ClassNotFoundException {
+    // TODO remove reflection if Spark 2.x support is dropped
+    if (package$.MODULE$.SPARK_VERSION().startsWith("2.")) {
+      Method spark2method = encoder.getClass().getMethod("toRow", Object.class);
+      return (InternalRow) spark2method.invoke(encoder, row);
```

Comment:
The problem here is ```hudi-spark2``` already depends on ```hudi-common``` and ```hudi-client```. Say if I create Spark2RowSerializer under ```hudi-spark2```, I also need to make ```hudi-client``` depends on ```hudi-spark2``` and as a result, it will bring a dependency loop.